### ISSUE TITLE
Domain Name Specification

### ISSUE THREAD

**Issue by mr-tron** [denizzzka/dianna2#5]
Я предлагаю взять уже готовую. 
https://wiki.namecoin.info/index.php?title=Domain_Name_Specification_2.0
Они там уже поучились на ошибках и выпустили вторую версию. И вообще она достаточно хорошо совместима с ICANN`овской системой. За исключением бесполезных тут SOA записей и подобного.

**Comment by denizzzka** [comment]
Я против поддержки всего многообразия DNS-записей: в некотором роде проект представляет собой самодеятельный корневой сервер DNS, а корневые серверы не хранят таких данных по понятным причинам.

К тому же, многие записи по ссылке предполагают более-менее регулярные их изменения в некоторых случаях, что вряд ли будет возможно в текущей схеме цепочки блоков.

Главной задачей считаю возможность дать ссылку на обыкновенный пользовательский DNS-сервер в условиях цензуры и всякого такого. Для этого "в обычном интернете", по большому счёту, нужны всего лишь две записи:

* NS (адрес сервера)
* DS (отпечаток ключа для DNSSEC)

Всё остальное выглядит излишеством, и проще настраивать другие записи через указание их в пользовательском DNS-сервере.

Также я против возможности указания поддоменов, так как эти возможности тоже реализуются в обычных DNS-серверах и их наличие может привести к неоднозначности их трактовки. То есть, предлагаю такой формат имени домена:
```
^([a-z0-9-]{0,62}[a-z0-9])?$
```

Плюс, конечно, нужны записи для резолва имён в tor, i2p, freenet и т.д.

**Comment by mr-tron** [comment]
Тоесть резолв имен для тор и прочих это ок, а для обычной сети - не ок?
90% обычных людей для доменов пользуется записью A, A для www и может mx. Зачем ради этого требовать отдельный сервер, если от этих 3 записей по 4 байта каждая(ну плюс немного обёрток) размер хранилища принципиально не изменится на фоне 32-байтных хэшей? Ну тоесть я ещё согласен, что сабдомены в хранилище не нужны. Хочешь сабдоменов - ставь ns записи и выноси всё на сторонние сервера по обычному протоколу dns. 
Плюс ты сам сказал, что это просто универсальное хранилище. ты же не станешь ограничивать то что люди туда хотят записать? ты можешь предложить набор записей на который будут обращать dns резолверы. причем желающие могут форкнуть и расширить список записей на которые смотреть. так почему бы просто заранее предложить уже некое опробированное решение, которое будет логично вписываться в имеющуюся схему?

P.S. регулярку я бы заменил, а то имя может начинаться на -
`^(?!.*-$)(?!-)[a-z0-9-]{1,63}$`

**Comment by denizzzka** [comment]
> Тоесть резолв имен для тор и прочих это ок, а для обычной сети - не ок?

для тора и прочих подобных по-другому не сделаешь

дело не в размере хранилища, а в невозможности быстрого обновления внутри хранилища и в поддержке передачи криптоключей стандартными методами. простейший ДНС-сервер не сложно поставить же

> так почему бы просто заранее предложить уже некое опробированное решение, которое будет логично вписываться в имеющуюся схему?

я же написал почему мне опробированное (в неймкоинах, про них же речь?) решение не нравится

> P.S. регулярку я бы заменил, а то имя может начинаться на -

пруф?

**Comment by mr-tron** [comment]
>дело не в размере хранилища, а в невозможности быстрого обновления внутри хранилища и в поддержке передачи криптоключей стандартными методами. 

не понимаю почему не возможно быстрое обновление. в плане оно конечно не моментальное - требует время сделать работу и чтобы изменения разбежались по сети. но так и сейчас днс до двух суток обновляется по кэшам. неизвестно что быстрее ещё. если кому-то надо реально быстро и часто менять - сделегируют на днс сервер. а куча народу записи меняет раз в несколько лет.
>простейший ДНС-сервер не сложно поставить же

У некоторых людей этот днс даже поставить некуда - нужен будет отдельный сервис днс хостинга с поддержкой домена в данной зоне.

>я же написал почему мне опробированное (в неймкоинах, про них же речь?) решение не нравится

Если ты про аргументы выше, то я вроде написал контродоводы.

>пруф?

https://regex101.com/r/eN7cG2/1
https://regex101.com/r/mZ1uN6/1

**Comment by denizzzka** [comment]
> не понимаю почему не возможно быстрое обновление. в плане оно конечно не моментальное - требует время сделать работу и чтобы изменения разбежались по сети. но так и сейчас днс до двух суток 

я предполагаю что тут изменения будут происходить в течении недели-двух.

и это не пруф - пруф это ссылка на RFC, где написнао что доменное имя может начинаться с "-"

**Comment by mr-tron** [comment]
>я предполагаю что тут изменения будут происходить в течении недели-двух.
почему? так долго работать на proof-of-work? изменения по сети распространятся быстро же.  часы в худшем случае. в битмесадже у меня переписка почти как в чате идёт - я даже думал про плагин для пиджина.

**Comment by denizzzka** [comment]
> почему? так долго работать на proof-of-work?

да. цена записи будет стремиться к $10 (цена домена второго уровня сейчас)

**Comment by mr-tron** [comment]
За 10$

**Comment by mr-tron** [comment]
За 10$ можно снять только хреновую виртуалку, которую игровой комп уделает за сутки. Если уж так оценивать, то смотреть на затраты электричества.  Хотя и тут можно посчитать только исходя из числа доменов, которые существуют в мире и смотреть по формуле роста сложности p-o-w какая будет сложность. опять же затраты на изменение не обязаны быть такими же как на создание.
Да и 10баксов это завышенная цена. зона .ru прекрасно живёт при цене 1.5 бакса за домен.

**Comment by denizzzka** [comment]
> За 10$ можно снять только хреновую виртуалку, которую игровой комп уделает за сутки. Если уж так оценивать, то смотреть на затраты электричества.

а, ну да.

в любом случае, срок добавления записей будет в районе нескольких дней - думаю, ты согласишься с этим?

в будущем можно сделать отдельный тип записей для ДНС и отдельный для изменений в них. но сначала надо решить вопрос с формулой - вдруг "дешёвые" записи об изменениях станут проблемой каким-то образом?

**Comment by mr-tron** [comment]
Я думаю срок добавления записи будет зависеть от формулы роста сложности. в биткоинах сложность растёт не ограничено, чтобы количество валюты, выпускаемой в сутки было стандартно. Но там потребность в деньгах и она не ограничена. Потребность в доменах на много ограниченнее и надо наверное на основе имеющихся доменных записей в обычном днс посчитать. вечерком набросаю. кстати это ещё одна из причин почему я предлагаю разною сложность. тогда хорошую часть работы по созданию сложности на себя возьмут титаны индустрии, которым нужны короткие домен и киберсквотеры. а васе пружинкину и его конторе "рога и копыта", которые довольствуются 8-9 значным доменом должно быть проще.

**Comment by denizzzka** [comment]
> Я думаю срок добавления записи будет зависеть от формулы роста сложности

то что я предлагаю это тоже формула сложности, но простая

> в биткоинах сложность растёт не ограничено

она и падать там умеет

> Но там потребность в деньгах и она не ограничена. 

ограничена затратами на их майнинг

> Потребность в доменах на много ограниченнее и надо наверное на основе имеющихся доменных записей в обычном днс посчитать. вечерком набросаю. кстати это ещё одна из причин почему я предлагаю разною сложность. тогда хорошую часть работы по созданию сложности на себя возьмут титаны индустрии, которым нужны короткие домен и киберсквотеры. а васе пружинкину и его конторе "рога и копыта", которые довольствуются 8-9 значным доменом должно быть проще.

у титанов не будет подавляющей "мощности" - любая мощность это деньги, а стоить требуемая мощность будет всё равно так и так по карману любому пользователю.

и я не понимаю почему титаны должны получить короткие домены. это несправедливым кажется. пусть каждый берёт то, что хочет, зачем усложнять? тем более, короткий домен не равнозначен хорошему дорогому домену

**Comment by denizzzka** [comment]
Пожалуй, A и AAAA записи добавим - это не сложно, но ничего страшного если их добавить не произойдёт. Если практика покажет, что они не нужны - уберём.

**Comment by mr-tron** [comment]
mx ещё тогда уж. это покроет потребности 90% пользователей.

**Comment by denizzzka** [comment]
не принесёт ли MX проблем с безопасностью?
